### DISCUSSION TITLE
mapping of accept header to route type

### DISCUSSION THREAD

**Discussion by pke** [sveltejs/kit#2842] [Q&A]
Is it possible to map request according to their `accept` header to svelte routes?

| Accept | route |
|-|-|
|`application/xml`| `routes/index.xml.js`|
|`text/xml`|`routes/index.xml.js`|
|`application/vnd.siren+json`|`routes/index.json.js`|

**Comment by CaptainCodeman** [comment]
IMO you may be better just writing an adapter that will handle the marshalling based on the http header, and use it to decorate the response that the endpoints already return.

You might even be able to add it to the hooks function rather than in each endpoint.

**Reply by pke** [reply]
Not sure I understand what you mean (new to svelte + sveltekit). I thought adapters come only into play when building the project like for vercel or node?

Which hooks function you are referring to?

**Comment by CaptainCodeman** [comment]
Sorry, I was meaning "adapter" as in classic ports-and-adapters architecture, completely forgetting that Svelte uses the term :) (although a similar concept / reason).

Basically, imagine having a function that you pass the request and response to, that your endpoint has already created as an object. This function could `switch` on the accept header and apply a different serialization as required.

The `hooks` function in SK is where you can handle requests (literally, "hook" into the request / response) and add some extra behavior. These are all concepts you'll see in many frameworks and languages under the heading "middleware".

https://kit.svelte.dev/docs#hooks
